web-console/knikubevirt/vm-details/vm-details.md

```diff
@@ -0,0 +1,114 @@
+# VM Details
+
+
+
+## Overview Tab
+
+![VM-overview](img/1-0-0.jpg)
```

Comment:
@cshinn Can you take a look at these charts? I believe these are using PF3 and we have updated charts in OpenShift to PF4. Also wondering if they belong above the standard detail info similar to how Nodes and Pods display charts at the top of the detail pages. 

Separate nit...should we swap status and namespace in the detail fields? Probably not important but figured it would match other resources more closely. Also we should change the project label to say namespace with the green NS badge :)

Reply:
> @lizsurette This is an oldie, but do you think the % uptime popover, accessed by clicking the info icon next to Status label, is necessary? I lean toward removing it along with the info icon entirely for discoverability, utility, and ambiguity reasons.

Are you suggesting we use the link here instead like we do on the table view? I like it, if so :)

> Unfortunately I don't remember why there are two IPs included in this design. It might be worth checking with devs if these IPs are different in any way. If not then this is fine, but if they are we could probably add sub-labels like Node Details pages do:

VMs can have multiple IPs (I think there would be one per NIC?) In any case we could probably organize them better and maybe sub-label them.

Reply:
@chart, I chatted with @andybraren a bit and it seems like those specific chart visuals were intended mostly as a placeholder. I expect that they would end up using the pf4 line graph like we do on the nodes pages etc. For consistency's sake, they should probably go at the top of the page as well as they are on the other pages

Reply:
> Are you suggesting we use the link here instead like we do on the table view? I like it, if so :)

@lizsurette I was suggesting removing it entirely, but I suppose the [Status Popover pattern](http://openshift.github.io/openshift-origin-design/web-console/4.0-designs/status/status#status-popovers) would be a natural place to include how long the VM has been running for, and when it was last powered on/off (if that's possible and we agree worth doing).

I'm just not sure what "uptime" means in a VM context and my googling isn't very fruitful. Is our "uptime" calculation how long the VM has been "Running" for? What would take it down from 100% to 97% in this example?

Reply:
> @lizsurette I was suggesting removing it entirely, but I suppose the [Status Popover pattern](http://openshift.github.io/openshift-origin-design/web-console/4.0-designs/status/status#status-popovers) would be a natural place to include how long the VM has been running for, and when it was last powered on/off (if that's possible and we agree worth doing).

Yeah I would assume we should be consistent with what we do in the table for this value.

> I'm just not sure what "uptime" means in a VM context and my googling isn't very fruitful. Is our "uptime" calculation how long the VM has been "Running" for? What would take it down from 100% to 97% in this example?

I think two values would be how long it's been running for and then how long it's been running in the last 7 days (%). I can't remember exactly where the original requirement came from though...it might be an oVirt carry over :)

Reply:
Here's an example of what those charts would look like (sorry for the delay. my cluster crashed on me yesterday)
![screencapture-console-openshift-console-apps-uxd-research-devcluster-openshift-k8s-cluster-nodes-ip-10-0-133-52-ap-northeast-1-compute-internal-2019-06-06-11_14_34](https://user-images.githubusercontent.com/3300756/59044578-6252db00-884c-11e9-97cf-8293b09e1cf0.png)